### ISSUE TITLE
NavigationPage.TitleView displays incorrectly on Windows

### ISSUE THREAD

**Issue by RWSchwabXXX** [dotnet/maui#16149]
### Description

When using a custom NavigationPage.TitleView compiled for Windows it displays incorrectly.  partial screenshots of identical code run on Windows and Andriod are below.  The windows version pushes everything to the start of the Title view and only extends the width of the encompasing grid to the largest child - rather than the width of the window.  Setting HorizontalOptions  on the grid (or any of the children) to fill do not change affect the display.

Windows
![image](https://github.com/dotnet/maui/assets/7891718/07d4c7f6-f089-4c3c-a3b6-ce61cb9715b2)

Android
![image](https://github.com/dotnet/maui/assets/7891718/bc0c785b-a945-4810-932c-61884381da80)

The ImageButton also displays differently, but that is of minor importance.

### Steps to Reproduce

Set the NavigationPage.TitleView in XAML for a ContentPage.

```
  <NavigationPage.TitleView>
    <Grid  >
      <Label Text="Home Page" HorizontalOptions="StartAndExpand" VerticalOptions="Center" TextColor="{AppThemeBinding Light={StaticResource Gray200}, Dark={StaticResource White}}" FontSize="Large"   />
      <Button Text="test" HorizontalOptions="Center" ImageSource="plus32.png" TextColor="White" />
      <ImageButton Command="{x:Binding TakePicture}" Source="camera_plus_32.png" HorizontalOptions="End" HeightRequest="32"/>
    </Grid>
  </NavigationPage.TitleView>
```
The content page is used to instantiate the main NavigationPage:
```
  public App(IServiceProvider services) {
    InitializeComponent();
    MainPage = new NavigationPage(services.GetService<Pages.PhotosHomePage>());
  }
```

### Link to public reproduction project repository

https://github.com/RWSchwabXXX/TitleviewProblemRepro

### Version with bug

7.0.49

### Last version that worked well

6.0

### Affected platforms

Windows

### Affected platform versions

windows10.0.19041.0

### Did you find any workaround?

Setting the WidthRequest of the grid within the TitleView helps, but it would need to be bound to the window width.

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.7.0 Preview 6.0. Repro on Windows 11(8.0.100-preview.6.23330.14), not repro on Android 13.0-API33 and iOS 16.4 with below Project:
[TitleviewProblemRepro.zip](https://github.com/dotnet/maui/files/12287652/TitleviewProblemRepro.zip)

**Comment by egvijayanand** [comment]
Yes, the NavigationBar TitleView does not expand to occupy the entire title bar even if the HorizontalOptions of the container control is set to Fill.

```xml
<NavigationPage.TitleView>
  <Grid BackgroundColor="Yellow"><Label Text="Home" /></Grid>
</NavigationPage.TitleView>
```

**Comment by ArminT28** [comment]
### **Workaround for TitleView width issue on Windows**

Hello!
I ran into the same TitleView width issue on Windows and wanted to share a small workaround that helps ensure the TitleView fills the full width of the window. This solution keeps most of the logic in XAML and uses a bit of code-behind to adjust the width dynamically based on the window size. The key part is using Dispatcher.Dispatch to safely set the width after the window has been initialized and to update it when the window size changes.

**Steps**
**_1.)_** In Class.xaml — Give the top-level element of your TitleView a name so it can be referenced in code:

```
<Grid x:Name="TitleGrid">
  <!-- Your title view content -->
</Grid>
```

**_2.)_** In Class.xaml.cs — Add a method that adjusts the width of the TitleGrid based on the window size:

```
private void AdjustTitleViewWidthOnWindows()
{
#if WINDOWS
    Dispatcher.Dispatch(() =>
    {
        var window = this.GetParentWindow();
        if (window != null)
        {
            TitleGrid.WidthRequest = window.Width;

            window.SizeChanged += (s, e) =>
            {
                TitleGrid.WidthRequest = window.Width;
            };
        }
    });
#endif
}
```

**_3.)_** Call this method in the constructor:

```
public PhotosHomePage()
{
    InitializeComponent();
    AdjustTitleViewWidthOnWindows();
}
```

**Screenshots**
Windows:
![Image](https://github.com/user-attachments/assets/5dfb37ca-47bd-4d4b-a017-272981d379ed)

Android:
![Image](https://github.com/user-attachments/assets/ae220012-72d7-41f1-9c76-67a4070faa9a)

I’ve created a [pull request with this fix and demo project here](https://github.com/RWSchwabXXX/TitleviewProblemRepro/pull/1) if you want to check it out or test it easily.

**Comment by BagavathiPerumal** [comment]
This issue has been verified in Visual Studio 17.14.9 preview 1.0 with the latest MAUI version 9.0.110. It can be reproduced on the Windows platform.
 
![Image](https://github.com/user-attachments/assets/b24bdd66-7b76-4420-8c5b-72689fa426a0)

**Comment by rdurfee** [comment]
This issue still occurs on .NET 10 with Microsoft.Maui.Controls 10.0.101
